kong/core/handler.lua

```diff
@@ -687,12 +687,19 @@ return {
   },
   header_filter = {
     before = function(ctx)
+      local var = ngx.var
+      local header = ngx.header
+
       if ctx.KONG_PROXIED then
         local now = get_now()
         -- time spent waiting for a response from upstream
         ctx.KONG_WAITING_TIME             = now - ctx.KONG_ACCESS_ENDED_AT
         ctx.KONG_HEADER_FILTER_STARTED_AT = now
       end
+
+      if singletons.configuration.headers[constants.HEADERS.UPSTREAM_STATUS] then
+        header[constants.HEADERS.UPSTREAM_STATUS]  = var.upstream_status
+      end
```

Comment:
`$upstream_status` can contain a comma-separated list of upstream status codes if several upstreams were selected. I see no test case covering this, so I am wondering if it an oversight?

Regardless, my understanding of this header is that it would return the status code of the _last_ upstream request (others are tracked by the healthchecks and transparent to the client). Thoughts?

Reply:
style: note the 2 spaces before `=` as well.

Reply:
In any case, this is an oversight.

I agree that this header should return the _last_ upstream status and multiple upstream retries for the same request should happen transparently.

I'm assuming that you agree with the above implementation and will work on it and ping you back.